File: src/presentation/interfaces/i_output_formatter.py
from abc import ABC, abstractmethod

class IOutputFormatter(ABC):
    @abstractmethod
    def format_help(self) -> str:
        pass

    @abstractmethod
    def format_welcome(self) -> str:
        pass

    @abstractmethod
    def format_model_switched(self, prompt: str) -> str:
        pass

    @abstractmethod
    def format_interpreted_input(self, raw: str, interpreted: str) -> str:
        pass

    @abstractmethod
    def format_response(self, domain_name: str, response: str) -> str:
        pass

    @abstractmethod
    def format_loading_models(self) -> str:
        pass

    @abstractmethod
    def format_elapsed_time(self, minutes: int, seconds: int) -> str:
        pass

    @abstractmethod
    def format_model_list(self, header: str, names: list[str], prefix: str = "- ") -> str:
        pass

    @abstractmethod
    def format_warning(self, message: str) -> str:
        pass

    @abstractmethod
    def format_error(self, message: str) -> str:
        pass

    @abstractmethod
    def format_goodbye(self) -> str:
        pass

File: angular-app/src/core/presentation/interfaces/i-output-formatter.ts
export interface IOutputFormatter {
  formatInputPrompt(providerName: string, modelName: string): string;
  formatHelp(): string;
  formatWelcome(): string;
  formatModelSwitched(prompt: string): string;
  formatInterpretedInput(raw: string, interpreted: string): string;
  formatResponse(domainName: string, response: string): string;
  formatLoadingModels(): string;
  formatElapsedTime(minutes: number, seconds: number): string;
  formatModelList(header: string, names: string[], prefix?: string): string;
  formatWarning(message: string): string;
  formatError(message: string): string;
  formatGoodbye(): string;
}
